src/ComplexVisual.js

```diff
@@ -245,7 +247,12 @@ class ComplexVisual extends Visual {
       }
     }
 
-    return desc;
+    return {
```

Comment:
Thank for the refactoring. I just ask for several changes which are listed below. Let's discuss :)

Does the situation exist when index has been changed inside the function? Can we use the field 'status: created, changed, error' instead of separate fields?

Reply:
Yes, index can be changed inside the function
I think I can rewrite this code using status field)